File: app/src/main/java/ie/dublinbuspal/android/data/local/dao/BusStopServiceDao.java
package ie.dublinbuspal.android.data.local.dao;


import androidx.room.Dao;
import androidx.room.Query;
import ie.dublinbuspal.android.data.local.DbInfo;
import ie.dublinbuspal.android.data.local.entity.BusStopService;

import java.util.List;

@Dao
public interface BusStopServiceDao extends BaseDao<BusStopService> {

    @Query(DbInfo.SELECT_BUS_STOP_SERVICE)
    BusStopService select(String stopId);

    @Query(DbInfo.SELECT_BUS_STOP_SERVICES)
    List<BusStopService> selectAll();

    @Query(DbInfo.DELETE_BUS_STOP_SERVICES)
    void deleteAll();

}

File: database/src/main/java/ie/dublinbuspal/database/dao/BusStopServiceDao.kt
package ie.dublinbuspal.database.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query
import ie.dublinbuspal.database.entity.BusStopServiceEntity
import io.reactivex.Maybe

@Dao
interface BusStopServiceDao {

    @Query("SELECT * FROM stop_services")
    fun selectAll(): Maybe<List<BusStopServiceEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(entities: List<BusStopServiceEntity>)

    @Query("DELETE FROM stop_services")
    fun deleteAll()

}
